crates/sui-node/src/lib.rs

```diff
@@ -2183,8 +2184,34 @@ impl SuiNode {
         checkpoint_store: &CheckpointStore,
         recovery: &ForkRecoveryConfig,
     ) -> Result<()> {
-        if recovery.checkpoint_overrides.is_empty() {
-            return Ok(());
+        // If configured overrides include a checkpoint whose locally computed digest mismatches,
+        // clear locally computed checkpoints from that sequence (inclusive).
+        for (seq, expected_digest_str) in &recovery.checkpoint_overrides {
+            let Ok(expected_digest) = CheckpointDigest::from_str(expected_digest_str) else {
+                anyhow::bail!(
+                    "Invalid checkpoint digest override for seq {}: {}",
+                    seq,
+                    expected_digest_str
+                );
+            };
+
+            if let Some(local_summary) = checkpoint_store.get_locally_computed_checkpoint(*seq)? {
+                let local_digest = sui_types::message_envelope::Message::digest(&local_summary);
+                if local_digest != expected_digest {
+                    info!(
+                        seq,
+                        local = %Self::get_digest_prefix(local_digest),
+                        expected = %Self::get_digest_prefix(expected_digest),
+                        "Fork recovery: clearing locally_computed_checkpoints from {} due to digest mismatch",
```

Comment:
Do you think we need any additional check to make sure that this is also the same checkpoint digest committed to the checkpoint watermarks table (`CheckpointWatermark::CheckpointForkDetected`)?
since watermarks table clear is happening independently below:
```
        if let Some((checkpoint_seq, checkpoint_digest)) =
            checkpoint_store.get_checkpoint_fork_detected()?
```